### ISSUE TITLE
allow different ports for streaming reception and API requests

### ISSUE THREAD

**Issue by disk91** [netdata/netdata#3830]
Server side config :

```
[api-key-79a5f]
    enabled = yes
    allow from = *
    default history = 3600
    default memory mode = ram
    health enabled by default = auto
    default postpone alarms on connect seconds = 60
    multiple connections = allow
```

Client Side config:
```
[stream]
    enabled = yes
   destination = tcp:145.XXX.XXX.XXX:8125
   api key =api-key-79a5f
   timeout seconds = 60
   default port = 8125
   buffer size bytes = 1024000
```

The communication is not working : 
log on the client side
2018-06-15 15:03:19: netdata INFO  : STREAM_SENDER[backup] : STREAM backup [send to tcp:145.239.156.140:8125]: initializing communication...
2018-06-15 15:03:19: netdata INFO  : STREAM_SENDER[backup] : STREAM backup [send to tcp:145.239.156.140:8125]: waiting response from remote netdata...
2018-06-15 15:04:19: netdata ERROR : STREAM_SENDER[backup] : STREAM backup [send to tcp:145.239.156.140:8125]: server is not replying properly (is it a netdata?).

Log on the server side
2018-06-15 15:05:29: netdata ERROR : STATSD_COLLECTOR[1] : STATSD: excess data '*/*' after value '*/*'
2018-06-15 15:06:34: netdata ERROR : STATSD_COLLECTOR[1] : STATSD: excess data 'netdata/1.10.0-288-g4b5c0c01_rolling' after value 'netdata/1.10.0-288-g4b5c0c01_rolling' (errno 11, Resource temporarily unavailable)

Any idea ?

**Comment by disk91** [comment]
Apparently the Stream interface is not pushing to statsd but on port 19999 (front)
Using port 19999 make it working.
This is a problem as:
1- I want to have different slave pushing data to a master
2- I do not want to expose dashboard to Internet directly but through a nginx.
Is there a way to use a different port for plugin.d API and HTTP Api ?

**Comment by Ferroin** [comment]
Not with netdata by itself.

However, with some effort in terms of configuration, you can achieve a similar result.  The stream interface doesn't work through nginx, so if you run that on a different port, you can use firewall rules to limit who can get to the netdata backend, and achieve a similar result.

**Comment by disk91** [comment]
So basically here is the solution I found and deployed:
- slaves are pushing to a master on tcp/19999 
- master have tcp/19999 open but restricted to the slave's IP + localhost
- on master a nginx, as a proxy for localhost, is offering a https / secured access to the netdata console

Honestly, the limit for this is to have to allow access to all the slaves vm, this is preventing automatic dynamic addition of new slaves (I mean when you do not use vlan for communication). This could be optimized by separating the backend API and the front end  on different ports.

**Comment by ktsaou** [comment]
> Honestly, the limit for this is to have to allow access to all the slaves vm, this is preventing automatic dynamic addition of new slaves (I mean when you do not use vlan for communication). This could be optimized by separating the backend API and the front end on different ports.

Yes, this is needed.

**Comment by Ehekatl** [comment]
same requirement here, we have data center all over the world, this is absolutely necessary for us.
right now we have to do slave -> headless proxy -> master to avoid compromise security

What I prefer is a slave -> muti-master setup, let prom to scrape all master on different port with ip restriction, leave port 19999 public with api key auth

**Comment by ktsaou** [comment]
@cakrit 	this is on our roadmap.

**Comment by ktsaou** [comment]
And I think I found another similar one.

**Comment by cakrit** [comment]
This will be resolved by #4969
